src/kvstore/raftex/RaftPart.cpp

```diff
@@ -616,11 +619,13 @@ void RaftPart::processAppendLogResponses(
         LogID lastLogId,
         LogID committedId,
         TermID prevLogTerm,
-        LogID prevLogId) {
+        LogID prevLogId,
+        std::vector<std::shared_ptr<Host>> hosts) {
     // Make sure majority have succeeded
     size_t numSucceeded = 0;
+    size_t index = 0;
     for (auto& res : resps) {
-        if (res.get_error_code() == cpp2::ErrorCode::SUCCEEDED) {
+        if (!hosts[index++]->isLearner() && res.get_error_code() == cpp2::ErrorCode::SUCCEEDED) {
```

Comment:
the response's index maybe not same with hosts

Reply:
Good catch